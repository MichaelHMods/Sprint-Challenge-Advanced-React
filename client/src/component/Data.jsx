import React from 'react';

import axios from 'axios';

//import component
import DataCard from './DataCard';
import NavBar from './NavBar';

class Data extends React.Component{
    state = {
        name: '',
        country: ' ',
        searches: '',
        player: [{}]
    }


//useEffect
componentDidMount() {
    axios
        .get('http://localhost:5000/api/players')
        .then(res => {
           
            this.setState({
                name: res.data.name,
                country: res.data.country,
                searches: res.data.searches,
                player:res.data
            });

        })
        .catch(err => console.log(err))
}; 

render(){

    return(
        
            <div className="soccer">
                <NavBar/>
                <div  >
                 <h1 data-testid="divtag" className="fancy-div">Soccer</h1>
                </div>
                {this.state.player.map((item, i) => <DataCard item={item} key={i}/>)}
           
        </div>
    )
}
};

export default Data;
