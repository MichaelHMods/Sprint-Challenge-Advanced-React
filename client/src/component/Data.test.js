import React from "react"
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Data from './Data';




it("renders", () => {
    const { asFragment } = render(<Data text="Soccer"/>); 
    expect(asFragment()).toMatchSnapshot();
});

it('inserts text in div', () => {
    const {getByTestId, getByText} = render(<Data/>);
    expect(getByTestId("divtag")).toHaveTextContent('soccer');
    expect(getByText('soccer')).toHaveClass('fancy-div');
})
