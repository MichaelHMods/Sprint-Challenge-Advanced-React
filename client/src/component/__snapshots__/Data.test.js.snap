// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="soccer"
  >
    <nav
      class="navbar"
    >
      <h1>
        Womens Soccer
      </h1>
      <div
        class="dark-mode__toggle"
      >
        <div
          class="toggle"
        />
      </div>
    </nav>
    <div>
      <h1
        class="fancy-div"
        data-testid="divtag"
      >
        Soccer
      </h1>
    </div>
    <div
      class="main-card"
    >
      <div
        class="inside-card"
      >
        <h1>
          Player: 
        </h1>
        <h2>
          Country or Origin:  
        </h2>
        <h5>
          Times searched on Google:  
        </h5>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
